ruff_cli/src/main.rs

```diff
@@ -160,9 +160,12 @@ fn check(args: CheckArgs, log_level: LogLevel) -> Result<ExitCode> {
     }
 
     if cli.add_noqa {
+        if !matches!(autofix, fix::FixMode::None) {
```

Comment:
This is a rare path, since you can't set `--fix` on the CLI with `--add-noqa`. But if the user's `pyproject.toml` has `fix = true`, we warn.